examples/Viewer.tsx

```diff
@@ -161,43 +157,62 @@ class Viewer extends React.Component<{}, ViewerState> {
         const data: DataTransfer = event.dataTransfer as DataTransfer;
 
         const files: FileList = input.files || data.files;
-        const filesArr: FileHTML[] = Array.from(files) as FileHTML[];
-        Promise.all(
-            filesArr.map((item) => {
-                return item.text();
-            })
-        ).then((parsedFiles) => {
-            const simulariumFileIndex = findIndex(filesArr, (file) =>
-                file.name.includes(".simularium")
-            );
-            let simulariumFile;
-            try {
-                simulariumFile = JSON.parse(parsedFiles[simulariumFileIndex]);
-            } catch (error) {
-                return this.onError(new FrontEndError(error.message));
-            }
-            const fileName = filesArr[simulariumFileIndex].name;
-            const geoAssets = filesArr.reduce((acc, cur, index) => {
-                if (index !== simulariumFileIndex) {
-                    acc[cur.name] = parsedFiles[index];
-                }
-                return acc;
-            }, {});
-            simulariumController
-                .changeFile({ simulariumFile, geoAssets }, fileName)
+        const filesArr: File[] = Array.from(files) as File[];
+
+        // try to find the simularium file first, then read it.
+        // then put all the other files as geoAssets.
+        const simulariumFileIndex = findIndex(filesArr, (file) =>
+            file.name.includes(".simularium")
+        );
+        try {
+            loadSimulariumFile(filesArr[simulariumFileIndex])
+                .then((simulariumFile: ISimulariumFile) =>
+                    // textify every file that was not the simularium file:
+                    Promise.all(
+                        filesArr.map((element, index) => {
+                            if (index !== simulariumFileIndex) {
+                                // is async call
+                                return element.text();
+                            } else {
+                                return simulariumFile;
```

Comment:
For clarity, it seems like you should return the async `loadSimulariumFile(element)` here, so then you're mapping over an array and no matter what you're returning a Promise instead of having this nested Promise all within another Promise.

Reply:
I'm not sure that would make it clearer to me but it will certainly make it more concise without any loss of clarity.  I'll make that change.